x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncTask.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.async;
+
+import java.util.Map;
+
+/**
+ * A task that supports asynchronous execution and safe temporary storage of results
+ */
+public interface AsyncTask {
```

Comment:
I cannot make AsyncTask an abstract class because both AsyncSearchTask  and AsyncEqlSearchTask implement this interface, but AsyncSearchTask needs to inherit from SearchTask and AsyncEqlSearchTask shouldn't do that.

Reply:
I see, I missed this and I see no way around it.